Native_UWP_V2/MainPage.xaml.cs

```diff
@@ -1,55 +1,49 @@
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
-using System.Net.Http;
+using System.Net.Http.Headers;
 using System.Threading.Tasks;
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Controls;
 
 // The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
 
-namespace active_directory_dotnet_native_uwp_v2
+namespace Native_UWP_V2
 {
     /// <summary>
     /// An empty page that can be used on its own or navigated to within a Frame.
     /// </summary>
     public sealed partial class MainPage : Page
     {
-        //Set the API Endpoint to Graph 'me' endpoint
-        readonly string graphAPIEndpoint = "https://graph.microsoft.com/v1.0/me";
-
         //Set the scope for API call to user.read
-        string[] scopes = new string[] { "user.read" };
+        private string[] scopes = new string[] { "user.read" };
 
-        // Below are the clientId (Application Id) of your app registration and the tenant information. 
+        // Below are the clientId (Application Id) of your app registration and the tenant information.
         // You have to replace:
         // - the content of ClientID with the Application Id for your app registration
-        // - Te content of Tenant by the information about the accounts allowed to sign-in in your application:
-        //   - For Work or School account in your org, use your tenant ID, or domain
-        //   - for any Work or School accounts, use organizations
-        //   - for any Work or School accounts, or Microsoft personal account, use common
-        //   - for Microsoft Personal account, use consumers
-        private const string ClientId = "4a1aa1d5-c567-49d0-ad0b-cd957a47f842";        
+        private const string ClientId = "[Application Id pasted from the application registration portal]";
 
-        public IPublicClientApplication PublicClientApp { get; } 
+        public IPublicClientApplication PublicClientApp { get; }
 
         public MainPage()
         {
             this.InitializeComponent();
 
-
             // To change from Microsoft public cloud to a national cloud, use another value of AzureCloudInstance
             PublicClientApp = PublicClientApplicationBuilder.Create(ClientId)
                 .WithAuthority("https://login.microsoftonline.com/common")
-                .WithLogging((level, message, containsPii) =>
-                {
-                    Debug.WriteLine($"MSAL: {level} {message} ");
-                }, LogLevel.Warning, enablePiiLogging: false, enableDefaultPlatformLogging: true)
-                .WithUseCorporateNetwork(true)
-                .WithRedirectUri(Constant.PublicClientRedirectUri)
-                .Build();                
+                .WithUseCorporateNetwork(false)
+                .WithRedirectUri("https://login.microsoftonline.com/common/oauth2/nativeclient")
```

Comment:
.WithDefaultRedirectUri() ?

Reply:
@jmprieur, I have made the changes using .WithDefaultRedirectUri() but it gives invalid_request error.
(invalid_request: The provided value for the input parameter 'redirect_uri' is not valid. The expected value is a URI which matches a redirect URI registered for this client application.)

Reply:
can you check in fiddler what it uses as reply URI?

---
In reply to: [430111578](https://github.com/Azure-Samples/active-directory-dotnet-native-uwp-v2/pull/38#discussion_r430111578) [](ancestors = 430111578)

Reply:
@kalyankrishna1, @jmprieur Updated Readme to include section containing [Alternate approach to use WithDefaultRedirectURI()](https://github.com/Azure-Samples/active-directory-dotnet-native-uwp-v2/pull/38/commits/cf3ca22902966c8c88399a228e43ae698f52c676).

Reply:
why can't `WithDefaultRedirectUri()` be used? It has less code, so would prefer to use it, as the same value is registered. If it's not working, we should open an issue in MSAL .NET

Reply:
@jennyf19, WithDefaultRedirectUri() is working but it requires more steps like user needs to get the callback URI for the app and add that in the Authentication blade in the portal.  
We had an internal discussion to put this approach in Readme as mentioned in the section [Alternate approach to use WithDefaultRedirectURI()(https://github.com/Azure-Samples/active-directory-dotnet-native-uwp-v2/pull/38/commits/cf3ca22902966c8c88399a228e43ae698f52c676).

Reply:
got it. thanks. :)